src/Contracts/Codec/CodecMatcherInterface.php

```diff
@@ -44,86 +45,83 @@ public function registerEncoder(MediaTypeInterface $mediaType, Closure $encoderC
      *
      * @param MediaTypeInterface $mediaType
      * @param Closure            $decoderClosure
-     *
-     * @return void
+     * @return $this
      */
     public function registerDecoder(MediaTypeInterface $mediaType, Closure $decoderClosure);
 
     /**
-     * Get encoder.
-     *
-     * @return EncoderInterface|null
+     * @param MediaTypeInterface $mediaType
+     * @param Closure $encoder
+     * @param Closure $decoder
+     * @return $this
      */
-    public function getEncoder();
+    public function registerBoth(MediaTypeInterface $mediaType, Closure $encoder, Closure $decoder);
 
     /**
-     * Set encoder.
+     * Register the default media type that should be used if match is not found.
      *
-     * @param EncoderInterface|Closure $encoder
+     * A default media type is required for instances where a fallback encoder is required. For example, if an error
+     * needs to be encoded but the error was caused by there not being an encoder match, a renderer will need a fallback
+     * encoder to use.
      *
-     * @return void
+     * @param MediaTypeInterface $mediaType
+     * @return $this
      */
-    public function setEncoder($encoder);
+    public function registerDefault(MediaTypeInterface $mediaType);
 
     /**
-     * Get decoder.
-     *
-     * @return DecoderInterface|null
+     * @return EncoderMatchInterface
      */
-    public function getDecoder();
+    public function getDefaultEncoder();
 
     /**
-     * Set decoder.
-     *
-     * @param DecoderInterface|Closure $decoder
+     * Get the matched encoder, or null if no match.
      *
-     * @return DecoderInterface
+     * @return EncoderMatchInterface|null
      */
-    public function setDecoder($decoder);
+    public function getEncoderMatch();
```

Comment:
This could be `getMatchedEncoder` instead. It's best that it's not `getEncoder` because it's not actually returning an encoder instance.